flockoff/utils/git_utils.py

```
@@ -0,0 +1,140 @@
+    Returns:
+        The URL of the remote as a string, or None if an error occurs.
+    """
+    try:
+        remote_url = subprocess.check_output(
+            ['git', 'remote', 'get-url', remote_name],
+            cwd=repo_path,
+            stderr=subprocess.PIPE
+        ).strip().decode('utf-8')
+        return remote_url.splitlines()[0]
+    except subprocess.CalledProcessError as e:
+        warnings.warn(f"Error getting URL for remote '{remote_name}': {e}. Ensure the remote exists and you are in a git repository.")
```

Ruff (0.8.2)

No explicit `stacklevel` keyword argument found

(B028)